src/widget/input/completion.rs

```diff
@@ -164,18 +185,81 @@ impl Completion {
         self.selection = Selection::None;
     }
 
-    pub fn process(&mut self, input: &str) {
-        let Some((head, rest)) = input.split_once('/') else {
+    /// Convert a list of User structs into Completion Entries
+    pub fn with_users(&mut self, users: Vec<User>) -> Self {
+        self.users = users
+            .iter()
+            .map(|u| {
+                // Let's make some eternal strings
+                // TODO! Is this bad?
+                let static_str = Box::leak(u.nickname().to_string().into_boxed_str());
+                Entry {
+                    title: static_str,
+                    args: vec![],
+                    completion_type: CompletionType::User,
+                }
+            })
+            .collect();
+        self.clone()
```

Comment:
Instead of `&mut self` in the signature, you can make it `self` and avoid this `clone` here. Then calling this would be `*self = self.with_users()`

Reply:
I changed this function to convert and return a slice of users into Entry's, rather than storing them against the Completion's state. Then we just operate on that list and build our filtered_entries list as per usual.
I hope I'm not misunderstanding, but as far as I understand, we're passing in a list of users to the widget on every `update`, so we need to look through the current users in a channel each time, in case any new ones have been added.